cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -0,0 +1,91 @@
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.services;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.common.NamespaceNotFoundException;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.security.StandardPermission;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import io.cdap.cdap.security.spi.authentication.AuthenticationContext;
+import io.cdap.cdap.security.spi.authorization.AccessEnforcer;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.TableNotFoundException;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import io.cdap.cdap.store.NamespaceTable;
+import io.cdap.cdap.store.RepositoryTable;
+
+/**
+ * Service that manages source control for repositories and applications.
+ * It exposes repository CRUD apis and source control tasks that do pull/pull/list applications in linked repository.
+ */
+public class SourceControlManagementService {
+
+  private final AccessEnforcer accessEnforcer;
+  private final AuthenticationContext authenticationContext;
+  private final TransactionRunner transactionRunner;
+
+  @Inject
+  public SourceControlManagementService(TransactionRunner transactionRunner,
+                                        AccessEnforcer accessEnforcer,
+                                        AuthenticationContext authenticationContext) {
+    this.transactionRunner = transactionRunner;
+    this.accessEnforcer = accessEnforcer;
+    this.authenticationContext = authenticationContext;
+  }
+
+  private RepositoryTable getRepositoryTable(StructuredTableContext context) throws TableNotFoundException {
+    return new RepositoryTable(context);
+  }
+
+  private NamespaceTable getNamespaceTable(StructuredTableContext context) throws TableNotFoundException {
+    return new NamespaceTable(context);
+  }
+
+  public void setRepository(NamespaceId namespace, RepositoryConfig repository) throws NamespaceNotFoundException {
+    accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+
+    TransactionRunners.run(transactionRunner, context -> {
+      NamespaceTable nsTable = getNamespaceTable(context);
+      if (nsTable.get(namespace) == null) {
+        throw new NamespaceNotFoundException(namespace);
+      }
+
+      RepositoryTable repoTable = getRepositoryTable(context);
+      repoTable.create(namespace, repository);
+    }, NamespaceNotFoundException.class);
+  }
+
+  public void deleteRepository(NamespaceId namespace) {
+    accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.DELETE);
+    
+    TransactionRunners.run(transactionRunner, context -> {
+      RepositoryTable repoTable = getRepositoryTable(context);
+      repoTable.delete(namespace);
+    });
+  }
+
+  public RepositoryConfig getRepository(NamespaceId namespace) {
```

Comment:
it's more consistent with our other services if this throws RepositoryNotFoundException if it doesn't exist.